WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -72,36 +84,36 @@ class MainActivity : AppCompatActivity() {
                         isMainFrame: Boolean,
                         replyProxy: JavaScriptReplyProxy
                     ) {
-                        sendAndroidMessage(message.data, mContext)
+                        onMessageReceived(message.data!!)
                     }
                 })
         } else {
             webview.addJavascriptInterface(object {
                 @JavascriptInterface
-                fun postMessage(message: String?) {
-                    sendAndroidMessage(message, mContext)
+                fun postMessage(message: String) {
+                    onMessageReceived(message)
                 }
             }, jsObjName)
         }
     }
 
     // Invokes native android sharing
-    val sendAndroidMessage = { message: String?, mContext: Context ->
+    private val onMessageReceived = { message: String ->
         val sendIntent: Intent = Intent().apply {
             action = Intent.ACTION_SEND
             putExtra(Intent.EXTRA_TEXT, message)
             type = "text/plain"
         }
         val shareIntent = Intent.createChooser(sendIntent, null)
-        startActivity(mContext, shareIntent, null)
+        startActivity(this, shareIntent, null)
```

Comment:
nit: `MainActivity.this` might be more readable. As-written it's not clear if `this` refers to MainActivity or to the `onMessageReceived` function (clearly it's the former since that's a Context, it's just not immediately obvious when reading).

Reply:
We can't access the context using dot notation but I can add a comment explaining what this is

Reply:
What if we used `applicationContext` instead of `this` then? (Test that it works like we expect, but I _think_ it should?)
Alternatively, maybe just doing like `val mainActivity = this` or `this@MainActivity`?

Reply:
applicationContext crashes the application but this@MainActivity works